### ISSUE TITLE
GCLK selection ID numbering discontinuity

### ISSUE THREAD

**Issue by VTO123** [drewfish/arduino-ZeroRegs#10]
There is a discontinuity in the ID numbers of Generic clock selection ID:s (SAMD21 datasheet table 15-5) which is not taken into account in the for-loop addressing of the ID:s in printZeroRegGCLK().

This library uses the following numbering (ZeroRegs.cpp, line 782 onward):
```
static const char ZeroRegsGCLK_NAME1F[] = "AC_DIG";
static const char ZeroRegsGCLK_NAME20[] = "AC_ANA";
static const char ZeroRegsGCLK_NAME21[] = "DAC";
static const char ZeroRegsGCLK_NAME22[] = "PTC";
```
while SAMD datasheet specifies
0x1F = AC_DIG
0x20 = --
0x21 = AC_ANA
0x22 = --
0x23 = DAC
0x24 = PTC

So the last few GCLK clock selection ID:s are addressed incorrectly.

**Comment by VTO123** [comment]
Ah, I may have mixed things up.
I have the Adafruit Trinket M0+, which uses the Atmel ATSAMD21E18 variant. Therefore I used Atmel datasheet.
This library is for Arduino Zero etc. which also use Atmel although ATSAMD21G18 variant, according to Arduino website.

_But_ Atmel was acquired by Microchip in 2016, and the ID numberings were apparently updated during Microchip times, and the new numbering is used in this library.

So this might not be an issue, at least for the new versions of the MCU.

**Comment by drewfish** [comment]
OK cool. I'm happy to fix anything. I think there are macros that distinguish between the exact chip being used (... being compiled for), e.g. distinguish between E18 and G18. This library could use those to set the right value for the variant.

I haven't looked at embedded stuff in a few years, so I haven't been keeping up with datasheet updates, etc.

**Comment by VTO123** [comment]
I got my code working by using the new datasheet, so I guess this library is OK.

Maybe it was just an error in the old version of the datasheet, as it would be weird from Atmel / Microchip to change the ID numbers instead of just adding new ones to fill the reserved slots, because that would break compatibility of old code on the new processor even if the product code stayed the same. Though I did not find any reference to changing the ID numbers in the revision history, which is a bit odd considering the histories of both documents match up to the old version.
And E18 and G18 should not differ in this matter, they just have different pin counts, and the datasheet covers both of them.

I think this issue can be closed, as it seems more related to the datasheet than this library.
